### PR TITLE
configure: Move optimization spec from build.sh

### PR DESCRIPTION
**Pull request by mborgerson**
And use -Og for debug builds.

### CODE REVIEW

File: configure [hunk 1]
```diff
@@ -5290,7 +5290,7 @@ if test "$skip_meson" = no; then
         -Ddocdir="$docdir" \
         -Dqemu_firmwarepath="$firmwarepath" \
         -Dqemu_suffix="$qemu_suffix" \
-        -Doptimization=$(if test "$debug" = yes; then echo 0; else echo 2; fi) \
+        -Doptimization=$(if test "$debug" = yes; then echo g; else echo 3; fi) \
```
Comment (GXTX) [inline configure:5293]: Perhaps `O2` would be best?

Comment (mborgerson) [reply configure:5293]: > Perhaps `O2` would be best?

Why would `O2` be best?

Comment (GXTX) [reply configure:5293]: Upstream default & more widely used as a sane release default.

Comment (mborgerson) [reply configure:5293]: xemu has been using -O3 for a long time now.

If you can provide a good technical reason why we should prefer O2 over O3, I'll consider it

**Pull request merged**